docs/userguide/searchspace.md

```diff
@@ -1,46 +1,271 @@
-# Search spaces
-
-## General information
+# Search Spaces
 
 The term "search space" refers to the domain of possible values for the parameters that are being optimized during a campaign. A search space represents the space within which BayBE explores and searches for the optimal solution. It is implemented via the [`SearchSpace`](baybe.searchspace.core.SearchSpace) class.
 
-A search space might be purely discrete, purely continuous, or hybrid. This is encoded by the ``type`` attribute (see [`SearchSpaceType`](baybe.searchspace.core.SearchSpaceType)). Furthermore, it is possible to restrict certain types of search spaces using [`Constraints`](./constraints.md).
+Note that a search space is not necessarily equal to the space of allowed measurements. That is, if configured properly, it is possible to add measurements to a campaign that are not part of the search space.
+
+In BayBE, a search space is a union of two (potentially empty) subspaces. The [`SubspaceDiscrete`](baybe.searchspace.discrete.SubspaceDiscrete) contains all discrete parameters, i.e., all parameters which inherit from [`DiscreteParameter`](baybe.parameters.base.DiscreteParameter). The [`SubspaceContinuous`](baybe.searchspace.continuous.SubspaceContinuous) contains all continuous parameters, i.e., all parameters which inherit from [´ContinuousParameter´](baybe.parameters.base.ContinuousParameter).
+
+Depending on which of the subspaces are non-empty, a `SearchSpace` has exactly one of the three following [`SearchSpaceType`](baybe.searchspace.core.SearchSpaceType):
+1. If `SubspaceDiscrete` is non-empty but `SubspaceContinuous` is empty, the `SearchSpace` is *discrete* and has the type [`SearchSpaceType.DISCRETE`](baybe.searchspace.core.SearchSpaceType.DISCRETE).
+2. If `SubspaceDiscrete` is empty but `SubspaceContinuous` is non-empty, the `SearchSpace` is *continuous* and has the type [`SearchSpaceType.CONTINUOUS`](baybe.searchspace.core.SearchSpaceType.CONTINUOUS).
+3. If both subspaces are non-empty, the `SearchSpace` is *hybrid* and has the type [`SearchSpaceType.HYBRID`](baybe.searchspace.core.SearchSpaceType.HYBRID).
```

Comment:
also these statements are written in a slighly redundant manner, ther sno need to say `is called X and has type baybeenum.X`, just pick one of the two